graylog2-server/src/main/java/org/graylog2/shared/buffers/processors/ProcessBufferProcessor.java

```diff
@@ -75,11 +77,13 @@ public ProcessBufferProcessor(MetricRegistry metricRegistry,
         incomingMessages = metricRegistry.meter(name(ProcessBufferProcessor.class, "incomingMessages"));
         outgoingMessages = metricRegistry.meter(name(ProcessBufferProcessor.class, "outgoingMessages"));
         processTime = metricRegistry.timer(name(ProcessBufferProcessor.class, "processTime"));
+        currentEvent = null;
     }
 
     @Override
     public void onEvent(MessageEvent event) throws Exception {
         try {
+            currentEvent = event;
```

Comment:
How about setting (and clearing) this in `#dispatchMessage()` instead? That way we only get the single message. The `MessageEvent` could contain one or more messages.

Reply:
I agree. The downside is that we wouldn't see problems within a `DecodingProcessor` but that's less unlikely.